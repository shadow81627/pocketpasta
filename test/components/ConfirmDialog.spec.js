import Component from '@/components/ConfirmDialog';
import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import BootstrapVuePlugin from 'bootstrap-vue';
import Vue from 'vue';

const localVue = createLocalVue();
let vuetify;

localVue.use(Vuex);
localVue.use(BootstrapVuePlugin);
Vue.use(Vuetify);

describe('ViewSwitcher', () => {
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const factory = () =>
    shallowMount(Component, {
      localVue,
      vuetify,
      stubs: {
        NuxtLink: RouterLinkStub,
      },
    });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
