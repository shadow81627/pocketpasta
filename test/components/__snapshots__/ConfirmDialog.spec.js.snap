// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSwitcher renders properly 1`] = `
"<v-dialog-stub opendelay=\\"0\\" closedelay=\\"0\\" attach=\\"\\" contentclass=\\"\\" maxwidth=\\"none\\" origin=\\"center center\\" retainfocus=\\"true\\" transition=\\"dialog-transition\\" width=\\"600\\">
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\">
    <v-card-title-stub class=\\"headline grey lighten-2\\"><span style=\\"word-break: normal;\\">Are you sure you want to delete this item?</span></v-card-title-stub>
    <v-divider-stub></v-divider-stub>
    <v-card-actions-stub>
      <v-container-stub tag=\\"div\\">
        <v-row-stub tag=\\"div\\" nogutters=\\"true\\" justify=\\"end\\">
          <v-col-stub cols=\\"auto\\" tag=\\"div\\">
            <v-btn-stub tag=\\"button\\" activeclass=\\"\\" large=\\"true\\" type=\\"button\\"><span>No</span> <span class=\\"d-none d-sm-inline\\">, don't I changed my mind</span></v-btn-stub>
          </v-col-stub>
          <v-col-stub cols=\\"auto\\" tag=\\"div\\" class=\\"pl-4\\">
            <v-btn-stub color=\\"error\\" tag=\\"button\\" activeclass=\\"\\" large=\\"true\\" type=\\"button\\"><span>Yes</span> <span class=\\"d-none d-sm-inline\\">, I am sure</span></v-btn-stub>
          </v-col-stub>
        </v-row-stub>
      </v-container-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>"
`;
